test/helpers/txs.ts

```diff
@@ -1,6 +1,32 @@
 import { ContractReceipt, ContractTransaction } from 'ethers'
 
-export async function waitForTx(tx: Promise<ContractTransaction>): Promise<ContractReceipt> {
+import { retry } from './async'
+
+export async function waitForTx(tx: Promise<ContractTransaction>, _confirmations?: number): Promise<ContractReceipt> {
   const resolvedTx = await tx
-  return await resolvedTx.wait()
+  const confirmations = _confirmations ?? chainIdToConfirmationsNeededForFinalization(resolvedTx.chainId)
+
+  // retry .wait b/c sometimes it fails for the first time (probably due to a problem with provider)
+  return await retry(() => resolvedTx.wait(confirmations))
+}
+
+function chainIdToConfirmationsNeededForFinalization(chainId: number): number {
+  const defaultWhenReorgsPossible = 3
+  const defaultForInstantFinality = 0
+
+  // covers mainnet and public testnets
+  if (
+    (chainId > 0 && chainId < 6) ||
+    chainId === 42 // kovan
+  ) {
+    return defaultWhenReorgsPossible
+  } else if (chainId === 69 || chainId === 10) {
+    // optimism needs *some* delay
+    return 1
+  } else if (chainId === 421611) {
+    // arbitrum needs *some* delay
+    return 1
+  } else {
+    return defaultForInstantFinality
```

Comment:
Shouldn't the default be 1? Waiting for zero confirmation is the same as not calling `wait()` I think.

Reply:
hmm maybe? I'll try changing this

Reply:
Good catch. I thought `1` will be stuck on local chain....